dataprofiler/data_readers/data_utils.py

```diff
@@ -187,10 +212,14 @@ def read_json_df(data_generator, selected_columns=None, read_in_string=False):
         k += 1
     if not lines and k:
         raise ValueError("No JSON data could be read from these data.")
-    return json_to_dataframe(lines, selected_columns, read_in_string)
+    return json_to_dataframe(cast(List[Dict], lines), selected_columns, read_in_string)
```

Comment:
instead of casting, ca we not define it on 191?

Reply:
The issue is that `unicode_to_str` returns `Union[str, List, Dict]`. So `lines` becomes `List[Union[str, List, Dict]]`. However, it seems like `unicode_to_str` in this case should only return `Dict` so I've added a conditional check that should always pass before appending to `list` to tell mypy it's always a `Dict`.

Reply:
i see, in the future I wonder if we can address the fact that we use the object_hook and then wrap the output as well. seems like redundancy that might fix the issue to begin with:
```
obj = unicode_to_str(
    json.loads(
        raw_line, object_hook=unicode_to_str, object_pairs_hook=OrderedDict
    ),
    ignore_dicts=True,
)
```